### PR TITLE
CNJR-2412: Set database connection pool size based on worker thread count

### PR DESCRIPTION
**Pull request by micahlee**
### Desired Outcome

The desired outcome of this PR is to prevent a database thread pool exhaustion in Conjur.

### Implemented Changes

The primary change in this PR is to set the maximum database connections in the connection pool based on the number of threads per web worker, to mitigate the opportunity for connection exhaustion or for a request to time out waiting for a DB connection.

By default, the number of connections allowed in the pool is 1.2x the number of threads.

In addition to the core change this PR makes a couple of quality of life improvements that may be reviewed by commit:

- Factors out the Sequel ORM settings to a separate Rails initializer.
- Adds basic input validation and error reporting for the web worker, thread, and connection pool settings.
- Fixes a couple of issues currently present in the Changelog.

### Connected Issue/Story

CyberArk internal issue ID: CNJR-2412

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

We do not currently have any test suites that can execute tests at the scale to exhibit the faulty behavior this PR addresses. This approach, however, was validated in the field, and the overall configuration approach is validated through our E2E test suites.

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: config/initializers/sequel.rb [hunk 1]
```diff
@@ -10,3 +10,22 @@ def write_id_to_json response, field
...
+  # after policy loads [cyberark/conjur#2584](https://github.com/cyberark/conjur/pull/2584)
+  # The [ShardedThreadedConnectionPool](https://www.rubydoc.info/github/jeremyevans/sequel/Sequel/ShardedThreadedConnectionPool) does support closing connections on-demand.
+  # Sequel is configured to use the ShardedThreadedConnectionPool by setting the servers configuration on
+  # the database connection [docs](https://www.rubydoc.info/github/jeremyevans/sequel/Sequel%2FShardedThreadedConnectionPool:servers)
+  config.sequel.servers = {}
+
+  # Whether to dump the schema after successful migrations.
+  # Defaults to false in production and test, true otherwise.
+  config.sequel.schema_dump = false
+end
```
Comment (imheresamir) [inline config/initializers/sequel.rb]: Nit: newline needed at EOF. Will not hold up approval

**Review by imheresamir — APPROVED** [review]
Couple of very minor nits. Reviewed and looks good to me.

**Comment by micahlee** [comment]
Thanks @imheresamir, that was definitely a goof during history cleanup. That is fixed now. Mind re-approving?

**Review by imheresamir — APPROVED** [review]

**Comment by UberKitten** [comment]
@micahlee What was the thinking behind using 1.2 connections per thread?

**Pull request merged**